### PR TITLE
Adds acquire specific address implementation in ipam service

### PR DESCRIPTION
**Pull request by nerdalert**
Implements AcquireSpecificIP in IPAMService resolving #90

Signed-off-by: Brent Salisbury <bsalisbu@redhat.com>

### CODE REVIEW

**Comment by nerdalert** [comment]
PTAL at your convenience @majst01. I tested it in the project we are using it for and it worked as advertised. Appreciate you taking a look, cheers!

**Review by majst01 — APPROVED** [review]

File: pkg/service/ipam-service.go [hunk 1]
```diff
@@ -141,10 +141,20 @@ func (i *IPAMService) ReleaseChildPrefix(ctx context.Context, req *connect.Reque
...
-	resp, err := i.ipamer.AcquireIP(ctx, req.Msg.PrefixCidr)
-	if err != nil {
-		i.log.Errorw("acquireip", "error", err)
-		return nil, connect.NewError(connect.CodeInvalidArgument, err)
+	var resp *goipam.IP
+	var err error
+	if req.Msg.Ip != nil {
+		resp, err = i.ipamer.AcquireSpecificIP(ctx, req.Msg.PrefixCidr, *req.Msg.Ip)
+		if err != nil {
+			i.log.Errorw("acquireip", "error", err)
```
Comment (vishnoianil) [inline pkg/service/ipam-service.go:149]: acquireip --> acquirespecificip?

**Comment by majst01** [comment]
> PTAL at your convenience @majst01. I tested it in the project we are using it for and it worked as advertised. Appreciate you taking a look, cheers!

Thanks, this was fast :-)

Do you mind telling me your use case ?

**Comment by nerdalert** [comment]
> Do you mind telling me your use case ?

For sure! We are doing a POC for edge/iot networking. Was funny, I shared it with another colleague in networking and he was also looking at it for one of their projects so keep up the great work! We will definitely contribute in the future. :tada:

**Comment by majst01** [comment]
CI was failing because it tried to push with contributors identity which is actually not supported in our ci setup.

**Pull request merged**